src/evo/mnhftx.cpp

```diff
@@ -116,6 +116,10 @@ bool CheckMNHFTx(const ChainstateManager& chainman, const llmq::CQuorumManager&
         return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-mnhf-version");
     }
 
+    if (!Params().IsValidMNActivation(mnhfTx.signal.versionBit, pindexPrev->GetMedianTimePast())) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-mnhf-non-ehf");
+    }
```

Comment:
6c6980832c8579c288988eadd6c25c739187c81a: could probably be it's own distinct PR?

Reply:
could not, because it's impossible (or very hard) to implement this functionality within v20 buried.